versioned_docs/version-8.7/self-managed/identity/user-guide/configuration/alternative-db.md

```diff
@@ -0,0 +1,200 @@
+
+3. If using docker or kubernetes, ensure that the folder with the library is properly mounted as a volume.
+
+<Tabs groupId="oracle-config" defaultValue="envVars" queryString values={
+[
+{label: 'Environment variables', value: 'envVars' },
+{label: 'values.yaml', value: 'valuesYaml' },
+{label: 'application.yaml', value: 'applicationYaml' },
+]}>
+<TabItem value="envVars">
+
+```sh
+SPRING_DATASOURCE_URL="jdbc:oracle:thin:@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=${IDENTITY_DATABASE_HOST:})(PORT=${IDENTITY_DATABASE_PORT:}))(CONNECT_DATA=(SERVICE_NAME=${IDENTITY_DATABASE_NAME:}))(SECURITY=(SSL_SERVER_CERT_DN=\"CN={CERT_CN}, O={CERT_ORG},L={..},ST={..},C={..}\")))"
+SPRING_DATASOURCE_DRIVER_CLASS_NAME=oracle.jdbc.OracleDriver
+SPRING_JPA_DATABASE=oracle
+JAVA_TOOL_OPTIONS=$JAVA_OPTS
+
+```
+
+</TabItem>
+<TabItem value="valuesYaml">
+
+```yaml
+identity:
+  keycloak:
+    enabled: false
+  externalDatabase:
+    enabled: true
+  # These three configuration options are added so that spring knows to connect to oracledb using it's client library
+  env:
+    - name: SPRING_DATASOURCE_URL
+      value: 'jdbc:oracle:thin:@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=${IDENTITY_DATABASE_HOST:})(PORT=${IDENTITY_DATABASE_PORT:}))(CONNECT_DATA=(SERVICE_NAME=${IDENTITY_DATABASE_NAME:}))(SECURITY=(SSL_SERVER_CERT_DN="CN={CERT_CN}, O={CERT_ORG},L={..},ST={..},C={..}")))'
+    - name: SPRING_DATASOURCE_DRIVER_CLASS_NAME
+      value: oracle.jdbc.OracleDriver
+    - name: JAVA_TOOL_OPTIONS
+      value: $JAVA_OPTS
+    - name: SPRING_JPA_DATABASE
+      value: oracle
+  # Overriding identity.command is required so that the new driver in /app will be loaded upon startup.
+  command:
+    - /bin/sh
+    - -c
+  # the wget command downloads the odbc driver, and the -cp option loads the jar in the classpath so that the module is loaded. It could be replaced with an initContainer or by loading a volume that already has the odbc driver inside.
+  args:
+    - |
+      wget https://download.oracle.com/otn-pub/otn_software/jdbc/237/ojdbc17.jar -O /app/ojdbc.jar
+      java -cp "/app/ojdbc.jar:/app/identity.jar" org.springframework.boot.loader.launch.JarLauncher
+  # Extra volumes are mounted for any TLS certs necessary for the database:
+  extraVolumeMounts:
+    - name: "keystore-secret"
+      secret:
+        secretName: "keystore-secret"
+  extraVolumes:
+    - name: "keystore-secret"
+      mountPath: "/usr/local/certificates"
+```
+
+</TabItem>
+<TabItem value="applicationYaml">
+
+```yaml
+spring:
+  datasource:
+    url: jdbc:oracle:thin:@(DESCRIPTION=(ADDRESS=(PROTOCOL=tcps)(HOST=${IDENTITY_DATABASE_HOST:})(PORT=${IDENTITY_DATABASE_PORT:}))(CONNECT_DATA=(SERVICE_NAME=${IDENTITY_DATABASE_NAME:}))(SECURITY=(SSL_SERVER_CERT_DN=\"CN={CERT_CN}, O={CERT_ORG},L={..},ST={..},C={..}\")))
+    driver-class-name: oracle.jdbc.OracleDriver
+  jpa:
+    database: oracle
+```
+
+</TabItem>
+</Tabs>
+
+### MSSQL
+
```

Comment:
:wrench: Add a note that, unlike Oracle, Identity comes prepackaged with the MSSQL drivers, so no need to fetch them before.

Reply:
717d24057